src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -512,7 +512,7 @@ private void addInstanceOrClassVar(DetailAST varDefAst) {
                 && !isPrivateInstanceVariable(varDefAst)) {
             final DetailAST ident = varDefAst.findFirstToken(TokenTypes.IDENT);
             final VariableDesc desc = new VariableDesc(ident.getText(),
-                    varDefAst.findFirstToken(TokenTypes.TYPE), findScopeOfVariable(varDefAst));
+                    null, null);
```

Comment:
Considering `VariableDesc` is a nested private class, can we introduce another constructor `VariableDesc(String name)` and remove those null arguments?

Reply:
@rdiachenko done
There is no such solution for new surviving mutation. 
https://github.com/hcoles/pitest/issues/1230

Reply:
done

Reply:
Why not ?
```
private VariableDesc(String name) {
            this.name = name;
        }
```

And it will avoid Checker problems mostlikely

Reply:
we can't https://github.com/checkstyle/checkstyle/blob/c7f3688f4380144bb712bee7ad6808cd075b327c/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java#L836-L843
this both are final :)

Reply:
Can we cheat Checker by setting null in such new constructor?

Reply:
@romani like this https://github.com/checkstyle/checkstyle/commit/8d315e7c8e5e36e243a5cbb54d7c6a0bac7178a1

not working

Reply:
```
private VariableDesc(String name, DetailAST typeAst, DetailAST scope) {
    this(name);
    this.typeAst = typeAst;
    this.scope = scope;
}

private VariableDesc(String name) {
    this.name = name;
}
```
Just a bold idea. Do we actually need to keep them final? Considering both constructors and only `name` being common to them, those other fields look like optional, in addition they are not changed anyway. What value gives us keeping them final specifically in the context of this class?

Reply:
@rdiachenko After the declaration of typeAst and scope. they are only used to log the violation and in extra typeAst is used  athttps://github.com/checkstyle/checkstyle/blob/fb97cb48cb5b3aa75fa629acf094648be2717403/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java#L1010-L1020 
In all the places there is no such operation where the actual value is tried to be changed.

I think the removal of the final will also not create such an issue.

Reply:
@Vyom-Yadav @rdiachenko @romani 
One more thing I have not covered yet all the mutation of unusedLocalVariableCheck 3 has been left and 2 of them are related to this similar kind of reason that we are not throwing violation for instance variables and class variables.

https://github.com/checkstyle/checkstyle/blob/fb97cb48cb5b3aa75fa629acf094648be2717403/config/pitest-suppressions/pitest-coding-2-suppressions.xml#L255-L271
which is for 
https://github.com/checkstyle/checkstyle/blob/fb97cb48cb5b3aa75fa629acf094648be2717403/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java#L1010-L1020
hence this method is also returning https://github.com/checkstyle/checkstyle/blob/fb97cb48cb5b3aa75fa629acf094648be2717403/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java#L1008
so if we go with constructor then we can simple make 
```
                final VariableDesc variableDesc = new VariableDesc(instVar.getName(),
                        instVar.getTypeAst(), updatedScope);
```
as
```
            final VariableDesc variableDesc = new VariableDesc(instVar.getName());
```

Reply:
> What value gives us keeping them final specifically in the context of this class?

I would vote for keeping them final. It's not about whether we change them or not, but a variable once declared has certain fixed quantities, like name, scope, and type.

Reply:
Actually, I also vote for a constructor.

Reply:
@romani @rdiachenko ping

Reply:
I don't see reply to https://github.com/checkstyle/checkstyle/pull/13145#discussion_r1230957417

Reply:
@romani I have tried what you say it leads to other new suppression and after solving it it leads to some other suppression I request you to not cover this here please if you want to check I can update the pr.

Reply:
@romani ping

Reply:
@romani ping